util/git/ssh.go

```diff
@@ -51,9 +58,19 @@ func (a *PublicKeysWithOptions) ClientConfig() (*ssh.ClientConfig, error) {
 	if len(a.KexAlgorithms) > 0 {
 		kexAlgos = a.KexAlgorithms
 	} else {
-		kexAlgos = DefaultSSHKeyExchangeAlgorithms
+		kexAlgos = getDefaultSSHKeyExchangeAlgorithms()
 	}
 	config := ssh.Config{KeyExchanges: kexAlgos}
 	opts := &ssh.ClientConfig{Config: config, User: a.User, Auth: []ssh.AuthMethod{ssh.PublicKeys(a.Signer)}}
 	return a.SetHostKeyCallback(opts)
 }
+
+// getDefaultSSHKeyExchangeAlgorithms returns the default key exchange algorithms to be used
+func getDefaultSSHKeyExchangeAlgorithms() []string {
+	if fips140.Enabled() {
+		log.Info("loading fips140 compliant default key exchange algorithms")
+		return SupportedFIPSCompliantSSHKeyExchangeAlgorithms
+	}
+	log.Info("loading default key exchange algorithms")
```

Comment:
I think these logs could get spammy in the end. If we need to print logs about FIPS, we should print them at the start of the repo server imho.